scipy/optimize/_differentialevolution.py

```diff
@@ -578,13 +595,20 @@ def __init__(self, func, bounds, args=(),
         # least 5 long
         self.num_population_members = max(5, popsize * self.parameter_count)
 
+        if isinstance(init, str) and (init == 'sobol'):
```

Comment:
why not just have `if init == 'sobol'`?

Reply:
Indeed!

Reply:
Actually I remember why I did this. Because if `init` is not a str but an array, the comparison fails. I will add a comment in the code.
```python
>       if init == 'sobol':
E       FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
```